From 4ae058ad614520301831ac4c955512f11e6ff3fa Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 25 Sep 2020 07:48:55 +0300
Subject: [PATCH] [SW-21029] goya2: dont skip PCIE bringup for secure boot

According to HW team, Boot Strap bit for PCIE_EN
(Bit 2 of mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) is
valid only in legacy mode, where BTL has performed
the PCIE Phy initialization.
In secure boot mode, this bit should not be checked
and ARC1 PPBoot should always bring up pcie.

Change-Id: Ia933c78822340c147b8c57d8bd899cad78767fb3
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 966dc69..548ecce 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -144,22 +144,9 @@
 	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
 
-/*
- * This logic copied from BTL PCIe section (#4)
- * In BTL it is called when DIP switch for PCIe enable is set
- * Now we wish to do this logic also here as a measure of security
- * It means that in new PCB's version by default PCIe won't be enabled
- * So without proper FW device won't be accessible through PCIe
- * Hence device is more secured
- */
 static void preboot_pcie_enable(void)
 {
-	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
 	u32 reset_cause;
-
-	/* Check if already done by BTL */
-	if (boot_strap & 0x2)
-		return;
 
 	/* During hard reset triggered by LKD keep PCIe intact */
 	reset_cause = regmap_readl(mmPREBOOT_PCIE_EN);
